src/Microsoft.VisualStudio.LanguageServices.Razor/Editor/VisualStudioDocumentTracker.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor.Editor
+{
+    public abstract class VisualStudioDocumentTracker
+    {
+        public abstract event EventHandler ContextChanged;
+
+        public abstract bool IsSupportedProject { get; }
```

Comment:
The primary purpose of the document tracker is to track what project we're associated with and *if* that project is a dotnet core project. If we end up getting our own content type then we don't need this.